src/main/resources/api/service_openapi.yaml

```diff
@@ -1792,23 +1792,86 @@ components:
           description: The canonical GCP id for this location. For example, \'us-east1\'.
           type: string
           required: true
-        machineType:
-          description: The Compute Engine machine type of this instance, see
-            https://cloud.google.com/compute/docs/machine-types
-          type: string
-          required: true
         postStartupScript:
           description: >-
             Path to a Bash script that automatically runs after a notebook instance fully boots up.
             The path must be a URL or Cloud Storage path (gs://path-to-file/file-name). If the path
             is a cloud storage path, the created VM instance's service account's credentials is used
             to access it.
           type: string
+        machineType:
+          description: The Compute Engine machine type of this instance, see
+            https://cloud.google.com/compute/docs/machine-types
+          type: string
+          required: true
+        acceleratorConfig:
+          $ref: '#/components/schemas/GcpAiNotebookInstanceAcceleratorConfig'
+        installGpuDriver:
+          description: >-
+            Whether the end user authorizes Google Cloud to install GPU driver on this instance. If
+            this field is empty or set to false, the GPU driver won't be installed. Only applicable
+            to instances with GPUs.
+          type: boolean
+        customGpuDriverPath:
+          description: >-
+            Specify a custom Cloud Storage path where the GPU driver is stored. If not specified,
+            we'll automatically choose from official GPU drivers.
+          type: string
+        bootDiskType:
+          description: >-
+            The type of the boot disk attached to this instance, defaults to standard persistent
+            disk (PD_STANDARD). See https://cloud.google.com/ai-platform/notebooks/docs/reference/rest/v1/projects.locations.instances#DiskType
+          type: string
+        bootDiskSizeGb:
+          description: >-
+            Input only. The size of the boot disk in GB attached to this instance, up to a maximum
+            of 64000 GB (64 TB). The minimum recommended value is 100 GB. If not specified, this
+            defaults to 100.
+          type: integer
+          format: int64
+        dataDiskType:
+          description: >-
+            Input only. The type of the data disk attached to this instance, defaults to standard
+            persistent disk (PD_STANDARD). See https://cloud.google.com/ai-platform/notebooks/docs/reference/rest/v1/projects.locations.instances#DiskType
+          type: string
+        dataDiskSizeGb:
+          description: >-
+            Input only. The size of the data disk in GB attached to this instance, up to a maximum
+            of 64000 GB (64 TB). If not specified, this defaults to 100.
+          type: integer
+          format: int64
+        labels:
```

Comment:
Previously, we've talked about using labels for spend tracking within Terra, which would mean users couldn't arbitrarily set these. Is there a defined usecase for exposing this?

Reply:
Yes, I would suggest leaving this off for now.

Reply:
SG, removed.